### ISSUE TITLE
To-do list

### ISSUE THREAD

**Issue by woothie** [Winter-Craft/wintercraft#1]
**Shit left to decide or do to get this boy up and running, in a roughly categorized order:**

## Phase 1 - Groundwork/planning:
The absolute basics of what we need to figure out before we do any real work.  
- [x] Decide on world size and server settings
  - Suggestion (Elite): 20000x20000 blocks, amplified biomes, all snow
- [x] Complete the [list of plugins](https://github.com/Winter-Craft/wintercraft/wiki)
- [x] Whip Woothie into getting a test server up and running
  - It works but my god is it ever awful. All configs are default and half the shit is broken.
  - [x] Establish latest Spigot version we can use before plugins start to -really- shit themselves. Compatability spreadsheet?
    - Use [PaperMC](https://papermc.io/) instead of Spigot? 
    - Paperclip 1.12.2 seems to be working alright with our selection of plugins.
- [x] Make a Discord server?
  - We're using Prox's Discord server, everyone is there and why would we make a new server if everyone is already around? (thank god, my list of servers is big enough already)

## Phase 2 - Security, permissions and making the police state:
Establishing security measures, figuring out permissions, testing plugins for exploitable or clearly overpowered mechanics, figuring out ways to moderate the server and its little blockmen in as painless a way as possible, etc.  
- [x] Gather some staff to keep the sheeple in place
- [x] Configure all the plugins. Welp.
- [x] Identify things/commands we want to restrict for players (permissions), make roles.
- [x] Make sure backup procedures are in place and decently fail-proof. Send at intervals to off-site?
- [ ] Establish easy way for staff to update, restart, start and stop the server when neccesary:
  - Plugin updater (also check if it's still compatible with our version)
  - Hard-kill if the server should hang
  - Simple status/start/stop/restart script

## Phase 3 - The Makening™:
Creating items, server-spesific mechanics, playtesting, etc.  
- [x] Find a good seed/world and pre-generate it out to the world borders (yes there are world borders)
- [x] Implement snow mechanics, see the [poll](https://www.strawpoll.me/16713248/r) for what the people decided. As of 26th of October, that was:
  - Snow biomes ouch (after 5-10 minutes, 10-20% change of taking half a heart. Slow and/or blind?
  - Snow storms more ouch (after ?? (no?) minutes, ??% chance of taking a full(?) heart. More slow? and blind?
  - Make sure this is all easily configurable and/or easily disabled, if we find out it's just a pain in the ass rather than a cool/immersive mechanic.
- [x] When/if the snow thing happens, have it have an effect?
  - Elite: Snow Biome will give occasional slowness or minor damage from wandering it without leather armour. If they are in an area with a ceiling  within 10 blocks and a bright light nearby they are safe. Snowstorm (snowing weather) is the same except faster, gives you blindness and leather armour won't do shit.
( yes they're pretty much identical, shut up I don't like deleting things)
- [x] Find a way to do that thing we talked about, the one with the snow
  - Custom plugin????
- [ ] Make heirlooms and whatnot by means of RPGItems
- [x] Do other RPG plugin/McMMO memes
- [ ] Find someone who has a hard-on for lore and can line up our world with the other /tg/craft iterations, maybe write this all down somewhere.
  - [ ] Establish list of factions
  - [ ] Establish list of gods
  - [ ] Establish list of races and their features
  - [ ] Vampire memes and their powers/weaknesses

## Phase 4 - Preparing for the storm:
Check everything five times over, make sure everything is fine and won't break horribly on a server with 30+ players running around like idiots trying to exploit everything and everyone. Send out announcements of imminent launch, when we know we're ready.
- [ ] Playtest plugins for stability once everything is done.
- [ ] Get the PR ready to draw in people.
- [x] Announce launch date when we know for fucking sure when we're 100% ready. And then add a week or so to that, just for good measure.

We might want to split these into separate issues, but it's nice to have a complete list somewhere to refer to. And it has checkboxes!
